File: AssetManagement.Application/Assets/GetAssets/GetAssetsResponse.cs
using AssetManagement.Domain.Entities.Assets;

namespace AssetManagement.Application.Assets.GetAssets
{
    public sealed record GetAssetsResponse
    (
        Guid AssetId,
        string AssetName,
        AssetType AssetType,
        AssetStatus AssetStatus,
        string Manufacturer,
        string Model,
        string SerialNumber,
        string? MacAddress,
        string? WifiMacAddress,
        string? Imei,
        string? OperatingSystem,
        string? OperatingSystemVersion,
        Guid AssignedStoreId,
        Guid? AssignedUserId,
        string? AssignedEmployeeName,
        DateTime CreatedAtUtc,
        Guid? AssignedVendorId,
        string? AssignedVendorName,
        string? RfidTagId
        );
}

File: Client/src/app/features/assets/data-access/models/get-assets-response.ts
import { AssetStatus } from "./asset-status";
import { AssetType } from "./asset-type";

export interface GetAssetsResponse {
  readonly assetId: string;
  readonly assetName: string;
  readonly assetType: AssetType;
  readonly assetStatus: AssetStatus;
  readonly manufacturer: string;
  readonly model: string;
  readonly serialNumber: string;
  readonly createdAtUtc: string;
  readonly macAddress: string | null;
  readonly wifiMacAddress: string | null;
  readonly imei: string | null;
  readonly operatingSystem: string | null;
  readonly operatingSystemVersion: string | null;
  readonly assignedStoreId: string;
  readonly assignedUserId: string | null;
  readonly assignedEmployeeName: string | null;
  readonly assignedVendorId: string | null;
  readonly assignedVendorName: string | null;
  readonly rfidTagId: string | null;
}
